kolibri/plugins/user/assets/src/routes.js

```diff
@@ -52,6 +52,9 @@ export default [
         if (store.getters.facilities.length > 1 && !store.state.facilityId) {
           // Go to FacilitySelect with whereToNext => SignUpPage
           const whereToNext = router.getRoute(ComponentMap.SIGN_IN);
+          if (to.query.next) {
+            whereToNext.query.next = to.query.next;
+          }
```

Comment:
This needs to go into the `query` of `route` below. When I get redirected, I just go to this URL, with no `?next` param

http://localhost:8000/en/user/#/facilities

```
          const route = {
            ...router.getRoute(ComponentMap.FACILITY_SELECT),
            params: { whereToNext },
			query: { to.query.next } (only if to.query.next is provided)
          };
```

Reply:
Did you create a second facility when you were testing this? If you have more than 1 facility, you will go to the `/facilities` route instead of the `/sign-in` route. Run this command in a second terminal to create a new facility (replace `user:pass` with your own credentials).

```
curl -X "POST" "http://localhost:8000/api/auth/facility/" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -u 'user:pass' \
     -d $'{
  "name": "Second Facility"
}'

```